src/grpc/infer_handler.cc

```diff
@@ -216,7 +247,7 @@ InferResponseStart(TRITONSERVER_ResponseAllocator* allocator, void* userp)
 TRITONSERVER_Error*
 SetInferenceRequestMetadata(
     TRITONSERVER_InferenceRequest* inference_request,
-    const inference::ModelInferRequest& request)
+    const inference::ModelInferRequest& request, StateParameters* state_params)
```

Comment:
do we follow the pattern of using pointers for in/out parameters universally - or could we use a reference here (avoid the null check later)

Reply:
Probably majority percentage is pointers for in/out, but reference are sometimes used, I'm happy to use a reference in smaller cases like here. In larger chunks, I think it can get harder to read / spot smells, as the mutable references may look like local variables at a quick glance.

Reply:
your choice - I like to avoid the null check if we can for internal apis - but more of a clarification on my part